core/src/main/scala/kafka/server/RaftReplicaManager.scala

```diff
@@ -251,6 +251,14 @@ class RaftReplicaManager(config: KafkaConfig,
               (Some(Partition(topicPartition, time, configRepository, this)), None)
           }
           partition.foreach { partition =>
+            builder.topicNameToId(partition.topic) match {
+              case Some(id) =>
+                if (!partition.checkOrSetTopicId(id, true))
```

Comment:
`s/true/usingRaft = true/` to make it clear what this boolean is?  Also, since this same section of code appears twice, maybe refactor it out into a helper method?